agent/php_error.c

```diff
@@ -561,12 +561,28 @@ static int nr_php_should_record_error(int type, const char* format TSRMLS_DC) {
     return 0;
   }
 
+  /*
+   * For a the following error codes:
+   * E_WARNING || E_CORE_WARNING || E_COMPILE_WARNING || E_USER_WARNING
+   * PHP triggers an exception if EH_THROW is toggled on and then immediately
+   * returns after throwing the exception. See for more info:
+   * https://github.com/php/php-src/blob/master/main/main.c In that case, we
+   * should not handle it, but we should exist and let the exception handler
+   * deal with it; otherwise, we could record an error even if an exception is
+   * caught.
+   */
+  if (EG(error_handling) == EH_THROW) {
+    if ((E_WARNING == type) || (E_CORE_WARNING == type)
+        || (E_COMPILE_WARNING == type) || (E_USER_WARNING == type)) {
```

Comment:
The list of error types for which the exception is thrown differs between PHPs - PHP 7.4: https://github.com/php/php-src/blob/PHP-7.4.17/main/main.c#L1229-L1258, PHP 8.0: https://github.com/php/php-src/blob/PHP-8.0.13/main/main.c#L1231-L1251. Shouldn't this list be conditionally compiled depending on PHP version?

Reply:
Please see: [ed0456a](https://github.com/newrelic/newrelic-php-agent/pull/876/commits/ed0456a7013e7b1c4735a219fd8295e168c5941a)